File: dungeons-26.1.2/src/main/java/fishmod/utils/NameList.java
package fishmod.utils;

import java.util.ArrayList;
import java.util.List;

/** Case-insensitive, comma-separated player-name list (used for party-action whitelists/blacklists). */
public final class NameList {
    private NameList() {}

    public static boolean contains(String csv, String name) {
        if (csv == null || csv.isBlank() || name == null) return false;
        for (String s : csv.split(",")) {
            if (s.trim().equalsIgnoreCase(name.trim())) return true;
        }
        return false;
    }

    public static String add(String csv, String name) {
        if (name == null || name.isBlank() || contains(csv, name)) return csv;
        List<String> names = toList(csv);
        names.add(name.trim());
        return String.join(",", names);
    }

    public static String remove(String csv, String name) {
        if (name == null) return csv;
        List<String> names = toList(csv);
        names.removeIf(s -> s.equalsIgnoreCase(name.trim()));
        return String.join(",", names);
    }

    public static List<String> toList(String csv) {
        List<String> out = new ArrayList<>();
        if (csv == null || csv.isBlank()) return out;
        for (String s : csv.split(",")) {
            String t = s.trim();
            if (!t.isEmpty()) out.add(t);
        }
        return out;
    }
}

File: dungeons-1.21.11/src/main/kotlin/fishmod/utils/NameList.kt
package fishmod.utils

/** Case-insensitive, comma-separated player-name list (used for party-action whitelists/blacklists). */
object NameList {

    @JvmStatic
    fun contains(csv: String?, name: String?): Boolean {
        if (csv == null || csv.isBlank() || name == null) return false
        for (s in csv.split(",")) {
            if (s.trim().equals(name.trim(), ignoreCase = true)) return true
        }
        return false
    }

    @JvmStatic
    fun add(csv: String?, name: String?): String? {
        if (name == null || name.isBlank() || contains(csv, name)) return csv
        val names = toList(csv)
        names.add(name.trim())
        return names.joinToString(",")
    }

    @JvmStatic
    fun remove(csv: String?, name: String?): String? {
        if (name == null) return csv
        val names = toList(csv)
        names.removeIf { it.equals(name.trim(), ignoreCase = true) }
        return names.joinToString(",")
    }

    @JvmStatic
    fun toList(csv: String?): ArrayList<String> {
        val out = ArrayList<String>()
        if (csv == null || csv.isBlank()) return out
        for (s in csv.split(",")) {
            val t = s.trim()
            if (t.isNotEmpty()) out.add(t)
        }
        return out
    }
}
